docs/architecture.md

```diff
@@ -0,0 +1,91 @@
+# Architecture
+
+The code review bot is not a single piece, but a distributed system using several providers.
+
+![](architecture.png)
+
+## Workflow
+
+1. For every new diff on [Mozilla Phabricator instance](https://phabricator.services.mozilla.com/), an Harbormaster build is created (more details on the [Phabricator page](phabricator.md)
+2. The build triggers an HTTP request towareds the Events system hosted on Heroku.
+3. The Heroku web instance parses the Phabricator payload and stores the relevant information in a Redis database
+4. A pool of Heroku workers poll the Redis database, one of them retrieve the stored build,
+5. That worker then fetches the stack of patches, and try to apply them locally,
+6. If the stack fails to apply, an error is reported directly on the Phabricator revision,
+7. If the stack applies cleanly, it's pushed on the Try server, exactcly like a developer would do,
+8. The Try server creates a decision task on Taskcluster,
+9. The Decision tasks creates followup tasks, with code analyzers, and a custom code-review task.
+10. The code analyzers run on the stack we produced, and list potential issues in a JSON file, that is stored and shared as a Taskcluster artifact,
+11. The final code-review task is triggered once all analyzers complete and just sends a Pulse message.
+12. The pulse messages triggers our own task hosted in a Taskcluster hook
+13. That task retrieves all the analyzers artifacts with issues, filters them, and updates the Phabricator build, reporting potential issues.
+14. That same task also publishes the issues on our backend hosted on Heroku.
+
+The whole workflow takes about 12 to 15 minutes on average for a given revision.
+
+## Events
```

Comment:
Maybe better to merge this section into events.md, so we don't duplicate info?

Reply:
Same with some of the following sections (e.g. Backend)

Reply:
Not sure about that, i made it as an intro towards these files, where the info is a bit mode developed

Reply:
The problem is that whenever we have to update the docs, we need to remember to update two documents instead of just one